core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -88,6 +88,10 @@ class AbstractModel:
         Hyperparameters that will be used by the model (can be search spaces instead of fixed values).
         If None, model defaults are used. This is identical to passing an empty dictionary.
     """
+    ag_key: str | None = None  # set to string value for subclasses for use in AutoGluon
+    ag_name: str | None = None  # set to string value for subclasses for use in AutoGluon
+    ag_priority: int = 0  # set to int value for subclasses for use in AutoGluon
+    ag_priority_by_problem_type: dict[str, int] = {}  # if not set, we fall back to ag_priority
```

Comment:
Nit: Setting a mutable object as the default value might lead to some problems in the future.

Reply:
Any thoughts on how to avoid this? We could move it to a property instead, but it would look less elegant.

Reply:
They only alternative idea that comes to mind is turning this into an instance attribute and calling `copy` during `__init__`. This also feels less elegant though.

Reply:
MappingProxyType looks reasonable, I've updated the PR to use it